src/exampleTypes/Geo.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import { Flex } from '../layout/Flex'
+import { exampleTypes } from 'contexture-client'
+import injectTreeNode from '../utils/injectTreeNode'
+import AsyncSelect from 'react-select/lib/Async'
```

Comment:
I wonder if we want to take a dependency on react-select, or if we should make in injectable 🤔 I'm open either way

Reply:
It is a pretty feature-packed component. In this case I am using the Async version but it comes with tons of stuff so IMO we should utilize it. Selectize ended up being good for us for a very long time and it did save good amount of dev hours. 

Not sure why we need it 'injectable' right now.

Reply:
I definitely think we should continue to use react-select, but I have concerns around bundling it with contexture react:

- Making it a direct dependency means that we'd need to start doing maintenance release bumping its version.
- If we use react-select in products that use contexture-react, we'd likely end up with it installed twice - and bundling it with contexture-react would increase our bundle size even for users who don't care about geo filters.
- We haven't taken on any dependencies on third-party components yet, but we do have precedence of injecting controls - everything from Modals to actual number inputs. It seems like we should continue to follow that convention.
- Even if we internally never use another component besides react-select, we should be mindful of the fact that this is an open source library that we'd like to see as widely adopted as possible. We don't need to build an abstraction layer over react-select - but we can trivially make it so that any component with the API we expect could work (in the same way the geo filter is aimed at HERE maps now instead of google, but as long as you satisfy the API it could really come from anywhere).

The first two points could be solved by making it a peer dependency, but not the second two.

Reply:
Ok updated with the AsyncSelect only used in the storyboard.